datatypes/mcs_datatype.cpp

```diff
@@ -66,6 +66,41 @@ int128_t TypeAttributesStd::decimal128FromString(const utils::NullString& value,
   return decimal128FromString(value.unsafeStringRef(), saturate);
 }
 
+// SQL parser checks that given `value` is in a valid format.
+// The first symbol can be `-`. The `value` can contain `.` symbol.
+void decimalPrecisionAndScale(const utils::NullString& value, int& precision, int& scale)
```

Comment:
IMHO it is better to return std::pair<precision, scale> from the f() b/c one doesn't need to watch out side effects that happens with refs inside `decimalPrecisionAndScale`